File: 658-find-k-closest-elements/658-find-k-closest-elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        ArrayList<Integer> list = new ArrayList<>();
        int start =  0 ;
        int end  = arr.length - 1;
        while((end - start) >= k)
        {
          int   dstart  = Math.abs(arr[start] - x);
          int   dend = Math.abs(arr[end] - x);
            
            
            if(dstart <= dend)
            {
                end--;
            }
            else
            {
                start++;
            }
        }
        
        while(start <= end)
        {
            list.add(arr[start++]);
        }
     return list;
    }
}

File: 658-find-k-closest-elements/658-find-k-closest-elements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        int start = 0;
        int end = arr.size()-1;
        vector<int> ans;
        while((end - start) >= k)
        {
            int newstart = abs(arr[start] - x);
            int newend = abs(arr[end] - x);
            if(newstart <= newend)
            {
                end--;
            }
            else
            {
                start++;
            }       
        }
         while(start <= end)
                ans.push_back(arr[start++]);
        return ans;
    }
};